// @linus_janns/webcomponents/dist/webcomponents.js
/**
 * @license
 * Copyright 2019 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
const G = window, $t = G.ShadowRoot && (G.ShadyCSS === void 0 || G.ShadyCSS.nativeShadow) && "adoptedStyleSheets" in Document.prototype && "replace" in CSSStyleSheet.prototype, _t = Symbol(), Tt = /* @__PURE__ */ new WeakMap();
let Lt = class {
  constructor(t, e, r) {
    if (this._$cssResult$ = !0, r !== _t)
      throw Error("CSSResult is not constructable. Use `unsafeCSS` or `css` instead.");
    this.cssText = t, this.t = e;
  }
  get styleSheet() {
    let t = this.o;
    const e = this.t;
    if ($t && t === void 0) {
      const r = e !== void 0 && e.length === 1;
      r && (t = Tt.get(e)), t === void 0 && ((this.o = t = new CSSStyleSheet()).replaceSync(this.cssText), r && Tt.set(e, t));
    }
    return t;
  }
  toString() {
    return this.cssText;
  }
};
const ee = (n) => new Lt(typeof n == "string" ? n : n + "", void 0, _t), re = (n, ...t) => {
  const e = n.length === 1 ? n[0] : t.reduce((r, a, i) => r + ((o) => {
    if (o._$cssResult$ === !0)
      return o.cssText;
    if (typeof o == "number")
      return o;
    throw Error("Value passed to 'css' function must be a 'css' function result: " + o + ". Use 'unsafeCSS' to pass non-literal values, but take care to ensure page security.");
  })(a) + n[i + 1], n[0]);
  return new Lt(e, n, _t);
}, ne = (n, t) => {
  $t ? n.adoptedStyleSheets = t.map((e) => e instanceof CSSStyleSheet ? e : e.styleSheet) : t.forEach((e) => {
    const r = document.createElement("style"), a = G.litNonce;
    a !== void 0 && r.setAttribute("nonce", a), r.textContent = e.cssText, n.appendChild(r);
  });
}, St = $t ? (n) => n : (n) => n instanceof CSSStyleSheet ? ((t) => {
  let e = "";
  for (const r of t.cssRules)
    e += r.cssText;
  return ee(e);
})(n) : n;
/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
var ct;
const J = window, Et = J.trustedTypes, ae = Et ? Et.emptyScript : "", At = J.reactiveElementPolyfillSupport, wt = { toAttribute(n, t) {
  switch (t) {
    case Boolean:
      n = n ? ae : null;
      break;
    case Object:
    case Array:
      n = n == null ? n : JSON.stringify(n);
  }
  return n;
}, fromAttribute(n, t) {
  let e = n;
  switch (t) {
    case Boolean:
      e = n !== null;
      break;
    case Number:
      e = n === null ? null : Number(n);
      break;
    case Object:
    case Array:
      try {
        e = JSON.parse(n);
      } catch {
        e = null;
      }
  }
  return e;
} }, It = (n, t) => t !== n && (t == t || n == n), ht = { attribute: !0, type: String, converter: wt, reflect: !1, hasChanged: It };
let P = class extends HTMLElement {
  constructor() {
    super(), this._$Ei = /* @__PURE__ */ new Map(), this.isUpdatePending = !1, this.hasUpdated = !1, this._$El = null, this.u();
  }
  static addInitializer(t) {
    var e;
    this.finalize(), ((e = this.h) !== null && e !== void 0 ? e : this.h = []).push(t);
  }
  static get observedAttributes() {
    this.finalize();
    const t = [];
    return this.elementProperties.forEach((e, r) => {
      const a = this._$Ep(r, e);
      a !== void 0 && (this._$Ev.set(a, r), t.push(a));
    }), t;
  }
  static createProperty(t, e = ht) {
    if (e.state && (e.attribute = !1), this.finalize(), this.elementProperties.set(t, e), !e.noAccessor && !this.prototype.hasOwnProperty(t)) {
      const r = typeof t == "symbol" ? Symbol() : "__" + t, a = this.getPropertyDescriptor(t, r, e);
      a !== void 0 && Object.defineProperty(this.prototype, t, a);
    }
  }
  static getPropertyDescriptor(t, e, r) {
    return { get() {
      return this[e];
    }, set(a) {
      const i = this[t];
      this[e] = a, this.requestUpdate(t, i, r);
    }, configurable: !0, enumerable: !0 };
  }
  static getPropertyOptions(t) {
    return this.elementProperties.get(t) || ht;
  }
  static finalize() {
    if (this.hasOwnProperty("finalized"))
      return !1;
    this.finalized = !0;
    const t = Object.getPrototypeOf(this);
    if (t.finalize(), t.h !== void 0 && (this.h = [...t.h]), this.elementProperties = new Map(t.elementProperties), this._$Ev = /* @__PURE__ */ new Map(), this.hasOwnProperty("properties")) {
      const e = this.properties, r = [...Object.getOwnPropertyNames(e), ...Object.getOwnPropertySymbols(e)];
      for (const a of r)
        this.createProperty(a, e[a]);
    }
    return this.elementStyles = this.finalizeStyles(this.styles), !0;
  }
  static finalizeStyles(t) {
    const e = [];
    if (Array.isArray(t)) {
      const r = new Set(t.flat(1 / 0).reverse());
      for (const a of r)
        e.unshift(St(a));
    } else
      t !== void 0 && e.push(St(t));
    return e;
  }
  static _$Ep(t, e) {
    const r = e.attribute;
    return r === !1 ? void 0 : typeof r == "string" ? r : typeof t == "string" ? t.toLowerCase() : void 0;
  }
  u() {
    var t;
    this._$E_ = new Promise((e) => this.enableUpdating = e), this._$AL = /* @__PURE__ */ new Map(), this._$Eg(), this.requestUpdate(), (t = this.constructor.h) === null || t === void 0 || t.forEach((e) => e(this));
  }
  addController(t) {
    var e, r;
    ((e = this._$ES) !== null && e !== void 0 ? e : this._$ES = []).push(t), this.renderRoot !== void 0 && this.isConnected && ((r = t.hostConnected) === null || r === void 0 || r.call(t));
  }
  removeController(t) {
    var e;
    (e = this._$ES) === null || e === void 0 || e.splice(this._$ES.indexOf(t) >>> 0, 1);
  }
  _$Eg() {
    this.constructor.elementProperties.forEach((t, e) => {
      this.hasOwnProperty(e) && (this._$Ei.set(e, this[e]), delete this[e]);
    });
  }
  createRenderRoot() {
    var t;
    const e = (t = this.shadowRoot) !== null && t !== void 0 ? t : this.attachShadow(this.constructor.shadowRootOptions);
    return ne(e, this.constructor.elementStyles), e;
  }
  connectedCallback() {
    var t;
    this.renderRoot === void 0 && (this.renderRoot = this.createRenderRoot()), this.enableUpdating(!0), (t = this._$ES) === null || t === void 0 || t.forEach((e) => {
      var r;
      return (r = e.hostConnected) === null || r === void 0 ? void 0 : r.call(e);
    });
  }
  enableUpdating(t) {
  }
  disconnectedCallback() {
    var t;
    (t = this._$ES) === null || t === void 0 || t.forEach((e) => {
      var r;
      return (r = e.hostDisconnected) === null || r === void 0 ? void 0 : r.call(e);
    });
  }
  attributeChangedCallback(t, e, r) {
    this._$AK(t, r);
  }
  _$EO(t, e, r = ht) {
    var a;
    const i = this.constructor._$Ep(t, r);
    if (i !== void 0 && r.reflect === !0) {
      const o = (((a = r.converter) === null || a === void 0 ? void 0 : a.toAttribute) !== void 0 ? r.converter : wt).toAttribute(e, r.type);
      this._$El = t, o == null ? this.removeAttribute(i) : this.setAttribute(i, o), this._$El = null;
    }
  }
  _$AK(t, e) {
    var r;
    const a = this.constructor, i = a._$Ev.get(t);
    if (i !== void 0 && this._$El !== i) {
      const o = a.getPropertyOptions(i), d = typeof o.converter == "function" ? { fromAttribute: o.converter } : ((r = o.converter) === null || r === void 0 ? void 0 : r.fromAttribute) !== void 0 ? o.converter : wt;
      this._$El = i, this[i] = d.fromAttribute(e, o.type), this._$El = null;
    }
  }
  requestUpdate(t, e, r) {
    let a = !0;
    t !== void 0 && (((r = r || this.constructor.getPropertyOptions(t)).hasChanged || It)(this[t], e) ? (this._$AL.has(t) || this._$AL.set(t, e), r.reflect === !0 && this._$El !== t && (this._$EC === void 0 && (this._$EC = /* @__PURE__ */ new Map()), this._$EC.set(t, r))) : a = !1), !this.isUpdatePending && a && (this._$E_ = this._$Ej());
  }
  async _$Ej() {
    this.isUpdatePending = !0;
    try {
      await this._$E_;
    } catch (e) {
      Promise.reject(e);
    }
    const t = this.scheduleUpdate();
    return t != null && await t, !this.isUpdatePending;
  }
  scheduleUpdate() {
    return this.performUpdate();
  }
  performUpdate() {
    var t;
    if (!this.isUpdatePending)
      return;
    this.hasUpdated, this._$Ei && (this._$Ei.forEach((a, i) => this[i] = a), this._$Ei = void 0);
    let e = !1;
    const r = this._$AL;
    try {
      e = this.shouldUpdate(r), e ? (this.willUpdate(r), (t = this._$ES) === null || t === void 0 || t.forEach((a) => {
        var i;
        return (i = a.hostUpdate) === null || i === void 0 ? void 0 : i.call(a);
      }), this.update(r)) : this._$Ek();
    } catch (a) {
      throw e = !1, this._$Ek(), a;
    }
    e && this._$AE(r);
  }
  willUpdate(t) {
  }
  _$AE(t) {
    var e;
    (e = this._$ES) === null || e === void 0 || e.forEach((r) => {
      var a;
      return (a = r.hostUpdated) === null || a === void 0 ? void 0 : a.call(r);
    }), this.hasUpdated || (this.hasUpdated = !0, this.firstUpdated(t)), this.updated(t);
  }
  _$Ek() {
    this._$AL = /* @__PURE__ */ new Map(), this.isUpdatePending = !1;
  }
  get updateComplete() {
    return this.getUpdateComplete();
  }
  getUpdateComplete() {
    return this._$E_;
  }
  shouldUpdate(t) {
    return !0;
  }
  update(t) {
    this._$EC !== void 0 && (this._$EC.forEach((e, r) => this._$EO(r, this[r], e)), this._$EC = void 0), this._$Ek();
  }
  updated(t) {
  }
  firstUpdated(t) {
  }
};
P.finalized = !0, P.elementProperties = /* @__PURE__ */ new Map(), P.elementStyles = [], P.shadowRootOptions = { mode: "open" }, At == null || At({ ReactiveElement: P }), ((ct = J.reactiveElementVersions) !== null && ct !== void 0 ? ct : J.reactiveElementVersions = []).push("1.6.1");
/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
var ft;
const K = window, M = K.trustedTypes, Ot = M ? M.createPolicy("lit-html", { createHTML: (n) => n }) : void 0, bt = "$lit$", C = `lit$${(Math.random() + "").slice(9)}$`, qt = "?" + C, ie = `<${qt}>`, D = document, F = () => D.createComment(""), j = (n) => n === null || typeof n != "object" && typeof n != "function", Ft = Array.isArray, oe = (n) => Ft(n) || typeof (n == null ? void 0 : n[Symbol.iterator]) == "function", vt = `[ 	
\f\r]`, L = /<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z][^>\s]*)|(\/?$))/g, Pt = /-->/g, kt = />/g, S = RegExp(`>|${vt}(?:([^\\s"'>=/]+)(${vt}*=${vt}*(?:[^ 	
\f\r"'\`<>=]|("|')|))|$)`, "g"), xt = /'/g, Mt = /"/g, jt = /^(?:script|style|textarea|title)$/i, se = (n) => (t, ...e) => ({ _$litType$: n, strings: t, values: e }), zt = se(1), U = Symbol.for("lit-noChange"), m = Symbol.for("lit-nothing"), Dt = /* @__PURE__ */ new WeakMap(), k = D.createTreeWalker(D, 129, null, !1), le = (n, t) => {
  const e = n.length - 1, r = [];
  let a, i = t === 2 ? "<svg>" : "", o = L;
  for (let s = 0; s < e; s++) {
    const l = n[s];
    let v, u, c = -1, f = 0;
    for (; f < l.length && (o.lastIndex = f, u = o.exec(l), u !== null); )
      f = o.lastIndex, o === L ? u[1] === "!--" ? o = Pt : u[1] !== void 0 ? o = kt : u[2] !== void 0 ? (jt.test(u[2]) && (a = RegExp("</" + u[2], "g")), o = S) : u[3] !== void 0 && (o = S) : o === S ? u[0] === ">" ? (o = a ?? L, c = -1) : u[1] === void 0 ? c = -2 : (c = o.lastIndex - u[2].length, v = u[1], o = u[3] === void 0 ? S : u[3] === '"' ? Mt : xt) : o === Mt || o === xt ? o = S : o === Pt || o === kt ? o = L : (o = S, a = void 0);
    const g = o === S && n[s + 1].startsWith("/>") ? " " : "";
    i += o === L ? l + ie : c >= 0 ? (r.push(v), l.slice(0, c) + bt + l.slice(c) + C + g) : l + C + (c === -2 ? (r.push(void 0), s) : g);
  }
  const d = i + (n[e] || "<?>") + (t === 2 ? "</svg>" : "");
  if (!Array.isArray(n) || !n.hasOwnProperty("raw"))
    throw Error("invalid template strings array");
  return [Ot !== void 0 ? Ot.createHTML(d) : d, r];
};
class z {
  constructor({ strings: t, _$litType$: e }, r) {
    let a;
    this.parts = [];
    let i = 0, o = 0;
    const d = t.length - 1, s = this.parts, [l, v] = le(t, e);
    if (this.el = z.createElement(l, r), k.currentNode = this.el.content, e === 2) {
      const u = this.el.content, c = u.firstChild;
      c.remove(), u.append(...c.childNodes);
    }
    for (; (a = k.nextNode()) !== null && s.length < d; ) {
      if (a.nodeType === 1) {
        if (a.hasAttributes()) {
          const u = [];
          for (const c of a.getAttributeNames())
            if (c.endsWith(bt) || c.startsWith(C)) {
              const f = v[o++];
              if (u.push(c), f !== void 0) {
                const g = a.getAttribute(f.toLowerCase() + bt).split(C), w = /([.?@])?(.*)/.exec(f);
                s.push({ type: 1, index: i, name: w[2], strings: g, ctor: w[1] === "." ? de : w[1] === "?" ? he : w[1] === "@" ? fe : nt });
              } else
                s.push({ type: 6, index: i });
            }
          for (const c of u)
            a.removeAttribute(c);
        }
        if (jt.test(a.tagName)) {
          const u = a.textContent.split(C), c = u.length - 1;
          if (c > 0) {
            a.textContent = M ? M.emptyScript : "";
            for (let f = 0; f < c; f++)
              a.append(u[f], F()), k.nextNode(), s.push({ type: 2, index: ++i });
            a.append(u[c], F());
          }
        }
      } else if (a.nodeType === 8)
        if (a.data === qt)
          s.push({ type: 2, index: i });
        else {
          let u = -1;
          for (; (u = a.data.indexOf(C, u + 1)) !== -1; )
            s.push({ type: 7, index: i }), u += C.length - 1;
        }
      i++;
    }
  }
  static createElement(t, e) {
    const r = D.createElement("template");
    return r.innerHTML = t, r;
  }
}
function W(n, t, e = n, r) {
  var a, i, o, d;
  if (t === U)
    return t;
  let s = r !== void 0 ? (a = e._$Co) === null || a === void 0 ? void 0 : a[r] : e._$Cl;
  const l = j(t) ? void 0 : t._$litDirective$;
  return (s == null ? void 0 : s.constructor) !== l && ((i = s == null ? void 0 : s._$AO) === null || i === void 0 || i.call(s, !1), l === void 0 ? s = void 0 : (s = new l(n), s._$AT(n, e, r)), r !== void 0 ? ((o = (d = e)._$Co) !== null && o !== void 0 ? o : d._$Co = [])[r] = s : e._$Cl = s), s !== void 0 && (t = W(n, s._$AS(n, t.values), s, r)), t;
}
class ue {
  constructor(t, e) {
    this._$AV = [], this._$AN = void 0, this._$AD = t, this._$AM = e;
  }
  get parentNode() {
    return this._$AM.parentNode;
  }
  get _$AU() {
    return this._$AM._$AU;
  }
  u(t) {
    var e;
    const { el: { content: r }, parts: a } = this._$AD, i = ((e = t == null ? void 0 : t.creationScope) !== null && e !== void 0 ? e : D).importNode(r, !0);
    k.currentNode = i;
    let o = k.nextNode(), d = 0, s = 0, l = a[0];
    for (; l !== void 0; ) {
      if (d === l.index) {
        let v;
        l.type === 2 ? v = new B(o, o.nextSibling, this, t) : l.type === 1 ? v = new l.ctor(o, l.name, l.strings, this, t) : l.type === 6 && (v = new ve(o, this, t)), this._$AV.push(v), l = a[++s];
      }
      d !== (l == null ? void 0 : l.index) && (o = k.nextNode(), d++);
    }
    return i;
  }
  v(t) {
    let e = 0;
    for (const r of this._$AV)
      r !== void 0 && (r.strings !== void 0 ? (r._$AI(t, r, e), e += r.strings.length - 2) : r._$AI(t[e])), e++;
  }
}
class B {
  constructor(t, e, r, a) {
    var i;
    this.type = 2, this._$AH = m, this._$AN = void 0, this._$AA = t, this._$AB = e, this._$AM = r, this.options = a, this._$Cp = (i = a == null ? void 0 : a.isConnected) === null || i === void 0 || i;
  }
  get _$AU() {
    var t, e;
    return (e = (t = this._$AM) === null || t === void 0 ? void 0 : t._$AU) !== null && e !== void 0 ? e : this._$Cp;
  }
  get parentNode() {
    let t = this._$AA.parentNode;
    const e = this._$AM;
    return e !== void 0 && (t == null ? void 0 : t.nodeType) === 11 && (t = e.parentNode), t;
  }
  get startNode() {
    return this._$AA;
  }
  get endNode() {
    return this._$AB;
  }
  _$AI(t, e = this) {
    t = W(this, t, e), j(t) ? t === m || t == null || t === "" ? (this._$AH !== m && this._$AR(), this._$AH = m) : t !== this._$AH && t !== U && this._(t) : t._$litType$ !== void 0 ? this.g(t) : t.nodeType !== void 0 ? this.$(t) : oe(t) ? this.T(t) : this._(t);
  }
  k(t) {
    return this._$AA.parentNode.insertBefore(t, this._$AB);
  }
  $(t) {
    this._$AH !== t && (this._$AR(), this._$AH = this.k(t));
  }
  _(t) {
    this._$AH !== m && j(this._$AH) ? this._$AA.nextSibling.data = t : this.$(D.createTextNode(t)), this._$AH = t;
  }
  g(t) {
    var e;
    const { values: r, _$litType$: a } = t, i = typeof a == "number" ? this._$AC(t) : (a.el === void 0 && (a.el = z.createElement(a.h, this.options)), a);
    if (((e = this._$AH) === null || e === void 0 ? void 0 : e._$AD) === i)
      this._$AH.v(r);
    else {
      const o = new ue(i, this), d = o.u(this.options);
      o.v(r), this.$(d), this._$AH = o;
    }
  }
  _$AC(t) {
    let e = Dt.get(t.strings);
    return e === void 0 && Dt.set(t.strings, e = new z(t)), e;
  }
  T(t) {
    Ft(this._$AH) || (this._$AH = [], this._$AR());
    const e = this._$AH;
    let r, a = 0;
    for (const i of t)
      a === e.length ? e.push(r = new B(this.k(F()), this.k(F()), this, this.options)) : r = e[a], r._$AI(i), a++;
    a < e.length && (this._$AR(r && r._$AB.nextSibling, a), e.length = a);
  }
  _$AR(t = this._$AA.nextSibling, e) {
    var r;
    for ((r = this._$AP) === null || r === void 0 || r.call(this, !1, !0, e); t && t !== this._$AB; ) {
      const a = t.nextSibling;
      t.remove(), t = a;
    }
  }
  setConnected(t) {
    var e;
    this._$AM === void 0 && (this._$Cp = t, (e = this._$AP) === null || e === void 0 || e.call(this, t));
  }
}
class nt {
  constructor(t, e, r, a, i) {
    this.type = 1, this._$AH = m, this._$AN = void 0, this.element = t, this.name = e, this._$AM = a, this.options = i, r.length > 2 || r[0] !== "" || r[1] !== "" ? (this._$AH = Array(r.length - 1).fill(new String()), this.strings = r) : this._$AH = m;
  }
  get tagName() {
    return this.element.tagName;
  }
  get _$AU() {
    return this._$AM._$AU;
  }
  _$AI(t, e = this, r, a) {
    const i = this.strings;
    let o = !1;
    if (i === void 0)
      t = W(this, t, e, 0), o = !j(t) || t !== this._$AH && t !== U, o && (this._$AH = t);
    else {
      const d = t;
      let s, l;
      for (t = i[0], s = 0; s < i.length - 1; s++)
        l = W(this, d[r + s], e, s), l === U && (l = this._$AH[s]), o || (o = !j(l) || l !== this._$AH[s]), l === m ? t = m : t !== m && (t += (l ?? "") + i[s + 1]), this._$AH[s] = l;
    }
    o && !a && this.j(t);
  }
  j(t) {
    t === m ? this.element.removeAttribute(this.name) : this.element.setAttribute(this.name, t ?? "");
  }
}
class de extends nt {
  constructor() {
    super(...arguments), this.type = 3;
  }
  j(t) {
    this.element[this.name] = t === m ? void 0 : t;
  }
}
const ce = M ? M.emptyScript : "";
class he extends nt {
  constructor() {
    super(...arguments), this.type = 4;
  }
  j(t) {
    t && t !== m ? this.element.setAttribute(this.name, ce) : this.element.removeAttribute(this.name);
  }
}
class fe extends nt {
  constructor(t, e, r, a, i) {
    super(t, e, r, a, i), this.type = 5;
  }
  _$AI(t, e = this) {
    var r;
    if ((t = (r = W(this, t, e, 0)) !== null && r !== void 0 ? r : m) === U)
      return;
    const a = this._$AH, i = t === m && a !== m || t.capture !== a.capture || t.once !== a.once || t.passive !== a.passive, o = t !== m && (a === m || i);
    i && this.element.removeEventListener(this.name, this, a), o && this.element.addEventListener(this.name, this, t), this._$AH = t;
  }
  handleEvent(t) {
    var e, r;
    typeof this._$AH == "function" ? this._$AH.call((r = (e = this.options) === null || e === void 0 ? void 0 : e.host) !== null && r !== void 0 ? r : this.element, t) : this._$AH.handleEvent(t);
  }
}
class ve {
  constructor(t, e, r) {
    this.element = t, this.type = 6, this._$AN = void 0, this._$AM = e, this.options = r;
  }
  get _$AU() {
    return this._$AM._$AU;
  }
  _$AI(t) {
    W(this, t);
  }
}
const Ut = K.litHtmlPolyfillSupport;
Ut == null || Ut(z, B), ((ft = K.litHtmlVersions) !== null && ft !== void 0 ? ft : K.litHtmlVersions = []).push("2.7.3");
const me = (n, t, e) => {
  var r, a;
  const i = (r = e == null ? void 0 : e.renderBefore) !== null && r !== void 0 ? r : t;
  let o = i._$litPart$;
  if (o === void 0) {
    const d = (a = e == null ? void 0 : e.renderBefore) !== null && a !== void 0 ? a : null;
    i._$litPart$ = o = new B(t.insertBefore(F(), d), d, void 0, e ?? {});
  }
  return o._$AI(n), o;
};
/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
var mt, gt;
class x extends P {
  constructor() {
    super(...arguments), this.renderOptions = { host: this }, this._$Do = void 0;
  }
  createRenderRoot() {
    var t, e;
    const r = super.createRenderRoot();
    return (t = (e = this.renderOptions).renderBefore) !== null && t !== void 0 || (e.renderBefore = r.firstChild), r;
  }
  update(t) {
    const e = this.render();
    this.hasUpdated || (this.renderOptions.isConnected = this.isConnected), super.update(t), this._$Do = me(e, this.renderRoot, this.renderOptions);
  }
  connectedCallback() {
    var t;
    super.connectedCallback(), (t = this._$Do) === null || t === void 0 || t.setConnected(!0);
  }
  disconnectedCallback() {
    var t;
    super.disconnectedCallback(), (t = this._$Do) === null || t === void 0 || t.setConnected(!1);
  }
  render() {
    return U;
  }
}
x.finalized = !0, x._$litElement$ = !0, (mt = globalThis.litElementHydrateSupport) === null || mt === void 0 || mt.call(globalThis, { LitElement: x });
const Wt = globalThis.litElementPolyfillSupport;
Wt == null || Wt({ LitElement: x });
((gt = globalThis.litElementVersions) !== null && gt !== void 0 ? gt : globalThis.litElementVersions = []).push("3.3.2");
/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
const Bt = (n) => (t) => typeof t == "function" ? ((e, r) => (customElements.define(e, r), r))(n, t) : ((e, r) => {
  const { kind: a, elements: i } = r;
  return { kind: a, elements: i, finisher(o) {
    customElements.define(e, o);
  } };
})(n, t);
/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
const ge = (n, t) => t.kind === "method" && t.descriptor && !("value" in t.descriptor) ? { ...t, finisher(e) {
  e.createProperty(t.key, n);
} } : { kind: "field", key: Symbol(), placement: "own", descriptor: {}, originalKey: t.key, initializer() {
  typeof t.initializer == "function" && (this[t.key] = t.initializer.call(this));
}, finisher(e) {
  e.createProperty(t.key, n);
} };
function Q(n) {
  return (t, e) => e !== void 0 ? ((r, a, i) => {
    a.constructor.createProperty(i, r);
  })(n, t, e) : ge(n, t);
}
/**
 * @license
 * Copyright 2021 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */
var pt;
((pt = window.HTMLSlotElement) === null || pt === void 0 ? void 0 : pt.prototype.assignedElements) != null;
function Z(n) {
  return Z = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(t) {
    return typeof t;
  } : function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, Z(n);
}
function A(n) {
  if (n === null || n === !0 || n === !1)
    return NaN;
  var t = Number(n);
  return isNaN(t) ? t : t < 0 ? Math.ceil(t) : Math.floor(t);
}
function p(n, t) {
  if (t.length < n)
    throw new TypeError(n + " argument" + (n > 1 ? "s" : "") + " required, but only " + t.length + " present");
}
function b(n) {
  p(1, arguments);
  var t = Object.prototype.toString.call(n);
  return n instanceof Date || Z(n) === "object" && t === "[object Date]" ? new Date(n.getTime()) : typeof n == "number" || t === "[object Number]" ? new Date(n) : ((typeof n == "string" || t === "[object String]") && typeof console < "u" && (console.warn("Starting with v2.0.0-beta.1 date-fns doesn't accept strings as date arguments. Please use `parseISO` to parse strings. See: https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md#string-arguments"), console.warn(new Error().stack)), /* @__PURE__ */ new Date(NaN));
}
function pe(n, t) {
  p(2, arguments);
  var e = b(n).getTime(), r = A(t);
  return new Date(e + r);
}
var ye = {};
function at() {
  return ye;
}
function we(n) {
  var t = new Date(Date.UTC(n.getFullYear(), n.getMonth(), n.getDate(), n.getHours(), n.getMinutes(), n.getSeconds(), n.getMilliseconds()));
  return t.setUTCFullYear(n.getFullYear()), n.getTime() - t.getTime();
}
function be(n) {
  return p(1, arguments), n instanceof Date || Z(n) === "object" && Object.prototype.toString.call(n) === "[object Date]";
}
function $e(n) {
  if (p(1, arguments), !be(n) && typeof n != "number")
    return !1;
  var t = b(n);
  return !isNaN(Number(t));
}
function _e(n, t) {
  p(2, arguments);
  var e = A(t);
  return pe(n, -e);
}
var Ce = 864e5;
function Te(n) {
  p(1, arguments);
  var t = b(n), e = t.getTime();
  t.setUTCMonth(0, 1), t.setUTCHours(0, 0, 0, 0);
  var r = t.getTime(), a = e - r;
  return Math.floor(a / Ce) + 1;
}
function tt(n) {
  p(1, arguments);
  var t = 1, e = b(n), r = e.getUTCDay(), a = (r < t ? 7 : 0) + r - t;
  return e.setUTCDate(e.getUTCDate() - a), e.setUTCHours(0, 0, 0, 0), e;
}
function Qt(n) {
  p(1, arguments);
  var t = b(n), e = t.getUTCFullYear(), r = /* @__PURE__ */ new Date(0);
  r.setUTCFullYear(e + 1, 0, 4), r.setUTCHours(0, 0, 0, 0);
  var a = tt(r), i = /* @__PURE__ */ new Date(0);
  i.setUTCFullYear(e, 0, 4), i.setUTCHours(0, 0, 0, 0);
  var o = tt(i);
  return t.getTime() >= a.getTime() ? e + 1 : t.getTime() >= o.getTime() ? e : e - 1;
}
function Se(n) {
  p(1, arguments);
  var t = Qt(n), e = /* @__PURE__ */ new Date(0);
  e.setUTCFullYear(t, 0, 4), e.setUTCHours(0, 0, 0, 0);
  var r = tt(e);
  return r;
}
var Ee = 6048e5;
function Ae(n) {
  p(1, arguments);
  var t = b(n), e = tt(t).getTime() - Se(t).getTime();
  return Math.round(e / Ee) + 1;
}
function et(n, t) {
  var e, r, a, i, o, d, s, l;
  p(1, arguments);
  var v = at(), u = A((e = (r = (a = (i = t == null ? void 0 : t.weekStartsOn) !== null && i !== void 0 ? i : t == null || (o = t.locale) === null || o === void 0 || (d = o.options) === null || d === void 0 ? void 0 : d.weekStartsOn) !== null && a !== void 0 ? a : v.weekStartsOn) !== null && r !== void 0 ? r : (s = v.locale) === null || s === void 0 || (l = s.options) === null || l === void 0 ? void 0 : l.weekStartsOn) !== null && e !== void 0 ? e : 0);
  if (!(u >= 0 && u <= 6))
    throw new RangeError("weekStartsOn must be between 0 and 6 inclusively");
  var c = b(n), f = c.getUTCDay(), g = (f < u ? 7 : 0) + f - u;
  return c.setUTCDate(c.getUTCDate() - g), c.setUTCHours(0, 0, 0, 0), c;
}
function Vt(n, t) {
  var e, r, a, i, o, d, s, l;
  p(1, arguments);
  var v = b(n), u = v.getUTCFullYear(), c = at(), f = A((e = (r = (a = (i = t == null ? void 0 : t.firstWeekContainsDate) !== null && i !== void 0 ? i : t == null || (o = t.locale) === null || o === void 0 || (d = o.options) === null || d === void 0 ? void 0 : d.firstWeekContainsDate) !== null && a !== void 0 ? a : c.firstWeekContainsDate) !== null && r !== void 0 ? r : (s = c.locale) === null || s === void 0 || (l = s.options) === null || l === void 0 ? void 0 : l.firstWeekContainsDate) !== null && e !== void 0 ? e : 1);
  if (!(f >= 1 && f <= 7))
    throw new RangeError("firstWeekContainsDate must be between 1 and 7 inclusively");
  var g = /* @__PURE__ */ new Date(0);
  g.setUTCFullYear(u + 1, 0, f), g.setUTCHours(0, 0, 0, 0);
  var w = et(g, t), T = /* @__PURE__ */ new Date(0);
  T.setUTCFullYear(u, 0, f), T.setUTCHours(0, 0, 0, 0);
  var H = et(T, t);
  return v.getTime() >= w.getTime() ? u + 1 : v.getTime() >= H.getTime() ? u : u - 1;
}
function Oe(n, t) {
  var e, r, a, i, o, d, s, l;
  p(1, arguments);
  var v = at(), u = A((e = (r = (a = (i = t == null ? void 0 : t.firstWeekContainsDate) !== null && i !== void 0 ? i : t == null || (o = t.locale) === null || o === void 0 || (d = o.options) === null || d === void 0 ? void 0 : d.firstWeekContainsDate) !== null && a !== void 0 ? a : v.firstWeekContainsDate) !== null && r !== void 0 ? r : (s = v.locale) === null || s === void 0 || (l = s.options) === null || l === void 0 ? void 0 : l.firstWeekContainsDate) !== null && e !== void 0 ? e : 1), c = Vt(n, t), f = /* @__PURE__ */ new Date(0);
  f.setUTCFullYear(c, 0, u), f.setUTCHours(0, 0, 0, 0);
  var g = et(f, t);
  return g;
}
var Pe = 6048e5;
function ke(n, t) {
  p(1, arguments);
  var e = b(n), r = et(e, t).getTime() - Oe(e, t).getTime();
  return Math.round(r / Pe) + 1;
}
function h(n, t) {
  for (var e = n < 0 ? "-" : "", r = Math.abs(n).toString(); r.length < t; )
    r = "0" + r;
  return e + r;
}
var xe = {
  // Year
  y: function(t, e) {
    var r = t.getUTCFullYear(), a = r > 0 ? r : 1 - r;
    return h(e === "yy" ? a % 100 : a, e.length);
  },
  // Month
  M: function(t, e) {
    var r = t.getUTCMonth();
    return e === "M" ? String(r + 1) : h(r + 1, 2);
  },
  // Day of the month
  d: function(t, e) {
    return h(t.getUTCDate(), e.length);
  },
  // AM or PM
  a: function(t, e) {
    var r = t.getUTCHours() / 12 >= 1 ? "pm" : "am";
    switch (e) {
      case "a":
      case "aa":
        return r.toUpperCase();
      case "aaa":
        return r;
      case "aaaaa":
        return r[0];
      case "aaaa":
      default:
        return r === "am" ? "a.m." : "p.m.";
    }
  },
  // Hour [1-12]
  h: function(t, e) {
    return h(t.getUTCHours() % 12 || 12, e.length);
  },
  // Hour [0-23]
  H: function(t, e) {
    return h(t.getUTCHours(), e.length);
  },
  // Minute
  m: function(t, e) {
    return h(t.getUTCMinutes(), e.length);
  },
  // Second
  s: function(t, e) {
    return h(t.getUTCSeconds(), e.length);
  },
  // Fraction of second
  S: function(t, e) {
    var r = e.length, a = t.getUTCMilliseconds(), i = Math.floor(a * Math.pow(10, r - 3));
    return h(i, e.length);
  }
};
const _ = xe;
var O = {
  am: "am",
  pm: "pm",
  midnight: "midnight",
  noon: "noon",
  morning: "morning",
  afternoon: "afternoon",
  evening: "evening",
  night: "night"
}, Me = {
  // Era
  G: function(t, e, r) {
    var a = t.getUTCFullYear() > 0 ? 1 : 0;
    switch (e) {
      case "G":
      case "GG":
      case "GGG":
        return r.era(a, {
          width: "abbreviated"
        });
      case "GGGGG":
        return r.era(a, {
          width: "narrow"
        });
      case "GGGG":
      default:
        return r.era(a, {
          width: "wide"
        });
    }
  },
  // Year
  y: function(t, e, r) {
    if (e === "yo") {
      var a = t.getUTCFullYear(), i = a > 0 ? a : 1 - a;
      return r.ordinalNumber(i, {
        unit: "year"
      });
    }
    return _.y(t, e);
  },
  // Local week-numbering year
  Y: function(t, e, r, a) {
    var i = Vt(t, a), o = i > 0 ? i : 1 - i;
    if (e === "YY") {
      var d = o % 100;
      return h(d, 2);
    }
    return e === "Yo" ? r.ordinalNumber(o, {
      unit: "year"
    }) : h(o, e.length);
  },
  // ISO week-numbering year
  R: function(t, e) {
    var r = Qt(t);
    return h(r, e.length);
  },
  // Extended year. This is a single number designating the year of this calendar system.
  // The main difference between `y` and `u` localizers are B.C. years:
  // | Year | `y` | `u` |
  // |------|-----|-----|
  // | AC 1 |   1 |   1 |
  // | BC 1 |   1 |   0 |
  // | BC 2 |   2 |  -1 |
  // Also `yy` always returns the last two digits of a year,
  // while `uu` pads single digit years to 2 characters and returns other years unchanged.
  u: function(t, e) {
    var r = t.getUTCFullYear();
    return h(r, e.length);
  },
  // Quarter
  Q: function(t, e, r) {
    var a = Math.ceil((t.getUTCMonth() + 1) / 3);
    switch (e) {
      case "Q":
        return String(a);
      case "QQ":
        return h(a, 2);
      case "Qo":
        return r.ordinalNumber(a, {
          unit: "quarter"
        });
      case "QQQ":
        return r.quarter(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "QQQQQ":
        return r.quarter(a, {
          width: "narrow",
          context: "formatting"
        });
      case "QQQQ":
      default:
        return r.quarter(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone quarter
  q: function(t, e, r) {
    var a = Math.ceil((t.getUTCMonth() + 1) / 3);
    switch (e) {
      case "q":
        return String(a);
      case "qq":
        return h(a, 2);
      case "qo":
        return r.ordinalNumber(a, {
          unit: "quarter"
        });
      case "qqq":
        return r.quarter(a, {
          width: "abbreviated",
          context: "standalone"
        });
      case "qqqqq":
        return r.quarter(a, {
          width: "narrow",
          context: "standalone"
        });
      case "qqqq":
      default:
        return r.quarter(a, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // Month
  M: function(t, e, r) {
    var a = t.getUTCMonth();
    switch (e) {
      case "M":
      case "MM":
        return _.M(t, e);
      case "Mo":
        return r.ordinalNumber(a + 1, {
          unit: "month"
        });
      case "MMM":
        return r.month(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "MMMMM":
        return r.month(a, {
          width: "narrow",
          context: "formatting"
        });
      case "MMMM":
      default:
        return r.month(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone month
  L: function(t, e, r) {
    var a = t.getUTCMonth();
    switch (e) {
      case "L":
        return String(a + 1);
      case "LL":
        return h(a + 1, 2);
      case "Lo":
        return r.ordinalNumber(a + 1, {
          unit: "month"
        });
      case "LLL":
        return r.month(a, {
          width: "abbreviated",
          context: "standalone"
        });
      case "LLLLL":
        return r.month(a, {
          width: "narrow",
          context: "standalone"
        });
      case "LLLL":
      default:
        return r.month(a, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // Local week of year
  w: function(t, e, r, a) {
    var i = ke(t, a);
    return e === "wo" ? r.ordinalNumber(i, {
      unit: "week"
    }) : h(i, e.length);
  },
  // ISO week of year
  I: function(t, e, r) {
    var a = Ae(t);
    return e === "Io" ? r.ordinalNumber(a, {
      unit: "week"
    }) : h(a, e.length);
  },
  // Day of the month
  d: function(t, e, r) {
    return e === "do" ? r.ordinalNumber(t.getUTCDate(), {
      unit: "date"
    }) : _.d(t, e);
  },
  // Day of year
  D: function(t, e, r) {
    var a = Te(t);
    return e === "Do" ? r.ordinalNumber(a, {
      unit: "dayOfYear"
    }) : h(a, e.length);
  },
  // Day of week
  E: function(t, e, r) {
    var a = t.getUTCDay();
    switch (e) {
      case "E":
      case "EE":
      case "EEE":
        return r.day(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "EEEEE":
        return r.day(a, {
          width: "narrow",
          context: "formatting"
        });
      case "EEEEEE":
        return r.day(a, {
          width: "short",
          context: "formatting"
        });
      case "EEEE":
      default:
        return r.day(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Local day of week
  e: function(t, e, r, a) {
    var i = t.getUTCDay(), o = (i - a.weekStartsOn + 8) % 7 || 7;
    switch (e) {
      case "e":
        return String(o);
      case "ee":
        return h(o, 2);
      case "eo":
        return r.ordinalNumber(o, {
          unit: "day"
        });
      case "eee":
        return r.day(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "eeeee":
        return r.day(i, {
          width: "narrow",
          context: "formatting"
        });
      case "eeeeee":
        return r.day(i, {
          width: "short",
          context: "formatting"
        });
      case "eeee":
      default:
        return r.day(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone local day of week
  c: function(t, e, r, a) {
    var i = t.getUTCDay(), o = (i - a.weekStartsOn + 8) % 7 || 7;
    switch (e) {
      case "c":
        return String(o);
      case "cc":
        return h(o, e.length);
      case "co":
        return r.ordinalNumber(o, {
          unit: "day"
        });
      case "ccc":
        return r.day(i, {
          width: "abbreviated",
          context: "standalone"
        });
      case "ccccc":
        return r.day(i, {
          width: "narrow",
          context: "standalone"
        });
      case "cccccc":
        return r.day(i, {
          width: "short",
          context: "standalone"
        });
      case "cccc":
      default:
        return r.day(i, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // ISO day of week
  i: function(t, e, r) {
    var a = t.getUTCDay(), i = a === 0 ? 7 : a;
    switch (e) {
      case "i":
        return String(i);
      case "ii":
        return h(i, e.length);
      case "io":
        return r.ordinalNumber(i, {
          unit: "day"
        });
      case "iii":
        return r.day(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "iiiii":
        return r.day(a, {
          width: "narrow",
          context: "formatting"
        });
      case "iiiiii":
        return r.day(a, {
          width: "short",
          context: "formatting"
        });
      case "iiii":
      default:
        return r.day(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // AM or PM
  a: function(t, e, r) {
    var a = t.getUTCHours(), i = a / 12 >= 1 ? "pm" : "am";
    switch (e) {
      case "a":
      case "aa":
        return r.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "aaa":
        return r.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "aaaaa":
        return r.dayPeriod(i, {
          width: "narrow",
          context: "formatting"
        });
      case "aaaa":
      default:
        return r.dayPeriod(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // AM, PM, midnight, noon
  b: function(t, e, r) {
    var a = t.getUTCHours(), i;
    switch (a === 12 ? i = O.noon : a === 0 ? i = O.midnight : i = a / 12 >= 1 ? "pm" : "am", e) {
      case "b":
      case "bb":
        return r.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "bbb":
        return r.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "bbbbb":
        return r.dayPeriod(i, {
          width: "narrow",
          context: "formatting"
        });
      case "bbbb":
      default:
        return r.dayPeriod(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // in the morning, in the afternoon, in the evening, at night
  B: function(t, e, r) {
    var a = t.getUTCHours(), i;
    switch (a >= 17 ? i = O.evening : a >= 12 ? i = O.afternoon : a >= 4 ? i = O.morning : i = O.night, e) {
      case "B":
      case "BB":
      case "BBB":
        return r.dayPeriod(i, {
          width: "abbreviated",
          context: "formatting"
        });
      case "BBBBB":
        return r.dayPeriod(i, {
          width: "narrow",
          context: "formatting"
        });
      case "BBBB":
      default:
        return r.dayPeriod(i, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Hour [1-12]
  h: function(t, e, r) {
    if (e === "ho") {
      var a = t.getUTCHours() % 12;
      return a === 0 && (a = 12), r.ordinalNumber(a, {
        unit: "hour"
      });
    }
    return _.h(t, e);
  },
  // Hour [0-23]
  H: function(t, e, r) {
    return e === "Ho" ? r.ordinalNumber(t.getUTCHours(), {
      unit: "hour"
    }) : _.H(t, e);
  },
  // Hour [0-11]
  K: function(t, e, r) {
    var a = t.getUTCHours() % 12;
    return e === "Ko" ? r.ordinalNumber(a, {
      unit: "hour"
    }) : h(a, e.length);
  },
  // Hour [1-24]
  k: function(t, e, r) {
    var a = t.getUTCHours();
    return a === 0 && (a = 24), e === "ko" ? r.ordinalNumber(a, {
      unit: "hour"
    }) : h(a, e.length);
  },
  // Minute
  m: function(t, e, r) {
    return e === "mo" ? r.ordinalNumber(t.getUTCMinutes(), {
      unit: "minute"
    }) : _.m(t, e);
  },
  // Second
  s: function(t, e, r) {
    return e === "so" ? r.ordinalNumber(t.getUTCSeconds(), {
      unit: "second"
    }) : _.s(t, e);
  },
  // Fraction of second
  S: function(t, e) {
    return _.S(t, e);
  },
  // Timezone (ISO-8601. If offset is 0, output is always `'Z'`)
  X: function(t, e, r, a) {
    var i = a._originalDate || t, o = i.getTimezoneOffset();
    if (o === 0)
      return "Z";
    switch (e) {
      case "X":
        return Ht(o);
      case "XXXX":
      case "XX":
        return E(o);
      case "XXXXX":
      case "XXX":
      default:
        return E(o, ":");
    }
  },
  // Timezone (ISO-8601. If offset is 0, output is `'+00:00'` or equivalent)
  x: function(t, e, r, a) {
    var i = a._originalDate || t, o = i.getTimezoneOffset();
    switch (e) {
      case "x":
        return Ht(o);
      case "xxxx":
      case "xx":
        return E(o);
      case "xxxxx":
      case "xxx":
      default:
        return E(o, ":");
    }
  },
  // Timezone (GMT)
  O: function(t, e, r, a) {
    var i = a._originalDate || t, o = i.getTimezoneOffset();
    switch (e) {
      case "O":
      case "OO":
      case "OOO":
        return "GMT" + Nt(o, ":");
      case "OOOO":
      default:
        return "GMT" + E(o, ":");
    }
  },
  // Timezone (specific non-location)
  z: function(t, e, r, a) {
    var i = a._originalDate || t, o = i.getTimezoneOffset();
    switch (e) {
      case "z":
      case "zz":
      case "zzz":
        return "GMT" + Nt(o, ":");
      case "zzzz":
      default:
        return "GMT" + E(o, ":");
    }
  },
  // Seconds timestamp
  t: function(t, e, r, a) {
    var i = a._originalDate || t, o = Math.floor(i.getTime() / 1e3);
    return h(o, e.length);
  },
  // Milliseconds timestamp
  T: function(t, e, r, a) {
    var i = a._originalDate || t, o = i.getTime();
    return h(o, e.length);
  }
};
function Nt(n, t) {
  var e = n > 0 ? "-" : "+", r = Math.abs(n), a = Math.floor(r / 60), i = r % 60;
  if (i === 0)
    return e + String(a);
  var o = t || "";
  return e + String(a) + o + h(i, 2);
}
function Ht(n, t) {
  if (n % 60 === 0) {
    var e = n > 0 ? "-" : "+";
    return e + h(Math.abs(n) / 60, 2);
  }
  return E(n, t);
}
function E(n, t) {
  var e = t || "", r = n > 0 ? "-" : "+", a = Math.abs(n), i = h(Math.floor(a / 60), 2), o = h(a % 60, 2);
  return r + i + e + o;
}
const De = Me;
var Yt = function(t, e) {
  switch (t) {
    case "P":
      return e.date({
        width: "short"
      });
    case "PP":
      return e.date({
        width: "medium"
      });
    case "PPP":
      return e.date({
        width: "long"
      });
    case "PPPP":
    default:
      return e.date({
        width: "full"
      });
  }
}, Xt = function(t, e) {
  switch (t) {
    case "p":
      return e.time({
        width: "short"
      });
    case "pp":
      return e.time({
        width: "medium"
      });
    case "ppp":
      return e.time({
        width: "long"
      });
    case "pppp":
    default:
      return e.time({
        width: "full"
      });
  }
}, Ue = function(t, e) {
  var r = t.match(/(P+)(p+)?/) || [], a = r[1], i = r[2];
  if (!i)
    return Yt(t, e);
  var o;
  switch (a) {
    case "P":
      o = e.dateTime({
        width: "short"
      });
      break;
    case "PP":
      o = e.dateTime({
        width: "medium"
      });
      break;
    case "PPP":
      o = e.dateTime({
        width: "long"
      });
      break;
    case "PPPP":
    default:
      o = e.dateTime({
        width: "full"
      });
      break;
  }
  return o.replace("{{date}}", Yt(a, e)).replace("{{time}}", Xt(i, e));
}, We = {
  p: Xt,
  P: Ue
};
const Ne = We;
var He = ["D", "DD"], Ye = ["YY", "YYYY"];
function Re(n) {
  return He.indexOf(n) !== -1;
}
function Le(n) {
  return Ye.indexOf(n) !== -1;
}
function Rt(n, t, e) {
  if (n === "YYYY")
    throw new RangeError("Use `yyyy` instead of `YYYY` (in `".concat(t, "`) for formatting years to the input `").concat(e, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
  if (n === "YY")
    throw new RangeError("Use `yy` instead of `YY` (in `".concat(t, "`) for formatting years to the input `").concat(e, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
  if (n === "D")
    throw new RangeError("Use `d` instead of `D` (in `".concat(t, "`) for formatting days of the month to the input `").concat(e, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
  if (n === "DD")
    throw new RangeError("Use `dd` instead of `DD` (in `".concat(t, "`) for formatting days of the month to the input `").concat(e, "`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md"));
}
var Ie = {
  lessThanXSeconds: {
    one: "less than a second",
    other: "less than {{count}} seconds"
  },
  xSeconds: {
    one: "1 second",
    other: "{{count}} seconds"
  },
  halfAMinute: "half a minute",
  lessThanXMinutes: {
    one: "less than a minute",
    other: "less than {{count}} minutes"
  },
  xMinutes: {
    one: "1 minute",
    other: "{{count}} minutes"
  },
  aboutXHours: {
    one: "about 1 hour",
    other: "about {{count}} hours"
  },
  xHours: {
    one: "1 hour",
    other: "{{count}} hours"
  },
  xDays: {
    one: "1 day",
    other: "{{count}} days"
  },
  aboutXWeeks: {
    one: "about 1 week",
    other: "about {{count}} weeks"
  },
  xWeeks: {
    one: "1 week",
    other: "{{count}} weeks"
  },
  aboutXMonths: {
    one: "about 1 month",
    other: "about {{count}} months"
  },
  xMonths: {
    one: "1 month",
    other: "{{count}} months"
  },
  aboutXYears: {
    one: "about 1 year",
    other: "about {{count}} years"
  },
  xYears: {
    one: "1 year",
    other: "{{count}} years"
  },
  overXYears: {
    one: "over 1 year",
    other: "over {{count}} years"
  },
  almostXYears: {
    one: "almost 1 year",
    other: "almost {{count}} years"
  }
}, qe = function(t, e, r) {
  var a, i = Ie[t];
  return typeof i == "string" ? a = i : e === 1 ? a = i.one : a = i.other.replace("{{count}}", e.toString()), r != null && r.addSuffix ? r.comparison && r.comparison > 0 ? "in " + a : a + " ago" : a;
};
const Fe = qe;
function yt(n) {
  return function() {
    var t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, e = t.width ? String(t.width) : n.defaultWidth, r = n.formats[e] || n.formats[n.defaultWidth];
    return r;
  };
}
var je = {
  full: "EEEE, MMMM do, y",
  long: "MMMM do, y",
  medium: "MMM d, y",
  short: "MM/dd/yyyy"
}, ze = {
  full: "h:mm:ss a zzzz",
  long: "h:mm:ss a z",
  medium: "h:mm:ss a",
  short: "h:mm a"
}, Be = {
  full: "{{date}} 'at' {{time}}",
  long: "{{date}} 'at' {{time}}",
  medium: "{{date}}, {{time}}",
  short: "{{date}}, {{time}}"
}, Qe = {
  date: yt({
    formats: je,
    defaultWidth: "full"
  }),
  time: yt({
    formats: ze,
    defaultWidth: "full"
  }),
  dateTime: yt({
    formats: Be,
    defaultWidth: "full"
  })
};
const Ve = Qe;
var Xe = {
  lastWeek: "'last' eeee 'at' p",
  yesterday: "'yesterday at' p",
  today: "'today at' p",
  tomorrow: "'tomorrow at' p",
  nextWeek: "eeee 'at' p",
  other: "P"
}, Ge = function(t, e, r, a) {
  return Xe[t];
};
const Je = Ge;
function I(n) {
  return function(t, e) {
    var r = e != null && e.context ? String(e.context) : "standalone", a;
    if (r === "formatting" && n.formattingValues) {
      var i = n.defaultFormattingWidth || n.defaultWidth, o = e != null && e.width ? String(e.width) : i;
      a = n.formattingValues[o] || n.formattingValues[i];
    } else {
      var d = n.defaultWidth, s = e != null && e.width ? String(e.width) : n.defaultWidth;
      a = n.values[s] || n.values[d];
    }
    var l = n.argumentCallback ? n.argumentCallback(t) : t;
    return a[l];
  };
}
var Ke = {
  narrow: ["B", "A"],
  abbreviated: ["BC", "AD"],
  wide: ["Before Christ", "Anno Domini"]
}, Ze = {
  narrow: ["1", "2", "3", "4"],
  abbreviated: ["Q1", "Q2", "Q3", "Q4"],
  wide: ["1st quarter", "2nd quarter", "3rd quarter", "4th quarter"]
}, tr = {
  narrow: ["J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D"],
  abbreviated: ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"],
  wide: ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
}, er = {
  narrow: ["S", "M", "T", "W", "T", "F", "S"],
  short: ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"],
  abbreviated: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
  wide: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
}, rr = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  }
}, nr = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  }
}, ar = function(t, e) {
  var r = Number(t), a = r % 100;
  if (a > 20 || a < 10)
    switch (a % 10) {
      case 1:
        return r + "st";
      case 2:
        return r + "nd";
      case 3:
        return r + "rd";
    }
  return r + "th";
}, ir = {
  ordinalNumber: ar,
  era: I({
    values: Ke,
    defaultWidth: "wide"
  }),
  quarter: I({
    values: Ze,
    defaultWidth: "wide",
    argumentCallback: function(t) {
      return t - 1;
    }
  }),
  month: I({
    values: tr,
    defaultWidth: "wide"
  }),
  day: I({
    values: er,
    defaultWidth: "wide"
  }),
  dayPeriod: I({
    values: rr,
    defaultWidth: "wide",
    formattingValues: nr,
    defaultFormattingWidth: "wide"
  })
};
const or = ir;
function q(n) {
  return function(t) {
    var e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, r = e.width, a = r && n.matchPatterns[r] || n.matchPatterns[n.defaultMatchWidth], i = t.match(a);
    if (!i)
      return null;
    var o = i[0], d = r && n.parsePatterns[r] || n.parsePatterns[n.defaultParseWidth], s = Array.isArray(d) ? lr(d, function(u) {
      return u.test(o);
    }) : sr(d, function(u) {
      return u.test(o);
    }), l;
    l = n.valueCallback ? n.valueCallback(s) : s, l = e.valueCallback ? e.valueCallback(l) : l;
    var v = t.slice(o.length);
    return {
      value: l,
      rest: v
    };
  };
}
function sr(n, t) {
  for (var e in n)
    if (n.hasOwnProperty(e) && t(n[e]))
      return e;
}
function lr(n, t) {
  for (var e = 0; e < n.length; e++)
    if (t(n[e]))
      return e;
}
function ur(n) {
  return function(t) {
    var e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, r = t.match(n.matchPattern);
    if (!r)
      return null;
    var a = r[0], i = t.match(n.parsePattern);
    if (!i)
      return null;
    var o = n.valueCallback ? n.valueCallback(i[0]) : i[0];
    o = e.valueCallback ? e.valueCallback(o) : o;
    var d = t.slice(a.length);
    return {
      value: o,
      rest: d
    };
  };
}
var dr = /^(\d+)(th|st|nd|rd)?/i, cr = /\d+/i, hr = {
  narrow: /^(b|a)/i,
  abbreviated: /^(b\.?\s?c\.?|b\.?\s?c\.?\s?e\.?|a\.?\s?d\.?|c\.?\s?e\.?)/i,
  wide: /^(before christ|before common era|anno domini|common era)/i
}, fr = {
  any: [/^b/i, /^(a|c)/i]
}, vr = {
  narrow: /^[1234]/i,
  abbreviated: /^q[1234]/i,
  wide: /^[1234](th|st|nd|rd)? quarter/i
}, mr = {
  any: [/1/i, /2/i, /3/i, /4/i]
}, gr = {
  narrow: /^[jfmasond]/i,
  abbreviated: /^(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)/i,
  wide: /^(january|february|march|april|may|june|july|august|september|october|november|december)/i
}, pr = {
  narrow: [/^j/i, /^f/i, /^m/i, /^a/i, /^m/i, /^j/i, /^j/i, /^a/i, /^s/i, /^o/i, /^n/i, /^d/i],
  any: [/^ja/i, /^f/i, /^mar/i, /^ap/i, /^may/i, /^jun/i, /^jul/i, /^au/i, /^s/i, /^o/i, /^n/i, /^d/i]
}, yr = {
  narrow: /^[smtwf]/i,
  short: /^(su|mo|tu|we|th|fr|sa)/i,
  abbreviated: /^(sun|mon|tue|wed|thu|fri|sat)/i,
  wide: /^(sunday|monday|tuesday|wednesday|thursday|friday|saturday)/i
}, wr = {
  narrow: [/^s/i, /^m/i, /^t/i, /^w/i, /^t/i, /^f/i, /^s/i],
  any: [/^su/i, /^m/i, /^tu/i, /^w/i, /^th/i, /^f/i, /^sa/i]
}, br = {
  narrow: /^(a|p|mi|n|(in the|at) (morning|afternoon|evening|night))/i,
  any: /^([ap]\.?\s?m\.?|midnight|noon|(in the|at) (morning|afternoon|evening|night))/i
}, $r = {
  any: {
    am: /^a/i,
    pm: /^p/i,
    midnight: /^mi/i,
    noon: /^no/i,
    morning: /morning/i,
    afternoon: /afternoon/i,
    evening: /evening/i,
    night: /night/i
  }
}, _r = {
  ordinalNumber: ur({
    matchPattern: dr,
    parsePattern: cr,
    valueCallback: function(t) {
      return parseInt(t, 10);
    }
  }),
  era: q({
    matchPatterns: hr,
    defaultMatchWidth: "wide",
    parsePatterns: fr,
    defaultParseWidth: "any"
  }),
  quarter: q({
    matchPatterns: vr,
    defaultMatchWidth: "wide",
    parsePatterns: mr,
    defaultParseWidth: "any",
    valueCallback: function(t) {
      return t + 1;
    }
  }),
  month: q({
    matchPatterns: gr,
    defaultMatchWidth: "wide",
    parsePatterns: pr,
    defaultParseWidth: "any"
  }),
  day: q({
    matchPatterns: yr,
    defaultMatchWidth: "wide",
    parsePatterns: wr,
    defaultParseWidth: "any"
  }),
  dayPeriod: q({
    matchPatterns: br,
    defaultMatchWidth: "any",
    parsePatterns: $r,
    defaultParseWidth: "any"
  })
};
const Cr = _r;
var Tr = {
  code: "en-US",
  formatDistance: Fe,
  formatLong: Ve,
  formatRelative: Je,
  localize: or,
  match: Cr,
  options: {
    weekStartsOn: 0,
    firstWeekContainsDate: 1
  }
};
const Sr = Tr;
var Er = /[yYQqMLwIdDecihHKkms]o|(\w)\1*|''|'(''|[^'])+('|$)|./g, Ar = /P+p+|P+|p+|''|'(''|[^'])+('|$)|./g, Or = /^'([^]*?)'?$/, Pr = /''/g, kr = /[a-zA-Z]/;
function xr(n, t, e) {
  var r, a, i, o, d, s, l, v, u, c, f, g, w, T, H, ot, st, lt;
  p(2, arguments);
  var Gt = String(t), Y = at(), R = (r = (a = e == null ? void 0 : e.locale) !== null && a !== void 0 ? a : Y.locale) !== null && r !== void 0 ? r : Sr, ut = A((i = (o = (d = (s = e == null ? void 0 : e.firstWeekContainsDate) !== null && s !== void 0 ? s : e == null || (l = e.locale) === null || l === void 0 || (v = l.options) === null || v === void 0 ? void 0 : v.firstWeekContainsDate) !== null && d !== void 0 ? d : Y.firstWeekContainsDate) !== null && o !== void 0 ? o : (u = Y.locale) === null || u === void 0 || (c = u.options) === null || c === void 0 ? void 0 : c.firstWeekContainsDate) !== null && i !== void 0 ? i : 1);
  if (!(ut >= 1 && ut <= 7))
    throw new RangeError("firstWeekContainsDate must be between 1 and 7 inclusively");
  var dt = A((f = (g = (w = (T = e == null ? void 0 : e.weekStartsOn) !== null && T !== void 0 ? T : e == null || (H = e.locale) === null || H === void 0 || (ot = H.options) === null || ot === void 0 ? void 0 : ot.weekStartsOn) !== null && w !== void 0 ? w : Y.weekStartsOn) !== null && g !== void 0 ? g : (st = Y.locale) === null || st === void 0 || (lt = st.options) === null || lt === void 0 ? void 0 : lt.weekStartsOn) !== null && f !== void 0 ? f : 0);
  if (!(dt >= 0 && dt <= 6))
    throw new RangeError("weekStartsOn must be between 0 and 6 inclusively");
  if (!R.localize)
    throw new RangeError("locale must contain localize property");
  if (!R.formatLong)
    throw new RangeError("locale must contain formatLong property");
  var V = b(n);
  if (!$e(V))
    throw new RangeError("Invalid time value");
  var Jt = we(V), Kt = _e(V, Jt), Zt = {
    firstWeekContainsDate: ut,
    weekStartsOn: dt,
    locale: R,
    _originalDate: V
  }, te = Gt.match(Ar).map(function(y) {
    var $ = y[0];
    if ($ === "p" || $ === "P") {
      var X = Ne[$];
      return X(y, R.formatLong);
    }
    return y;
  }).join("").match(Er).map(function(y) {
    if (y === "''")
      return "'";
    var $ = y[0];
    if ($ === "'")
      return Mr(y);
    var X = De[$];
    if (X)
      return !(e != null && e.useAdditionalWeekYearTokens) && Le(y) && Rt(y, t, String(n)), !(e != null && e.useAdditionalDayOfYearTokens) && Re(y) && Rt(y, t, String(n)), X(Kt, y, R.localize, Zt);
    if ($.match(kr))
      throw new RangeError("Format string contains an unescaped latin alphabet character `" + $ + "`");
    return y;
  }).join("");
  return te;
}
function Mr(n) {
  var t = n.match(Or);
  return t ? t[1].replace(Pr, "'") : n;
}
var Dr = Object.defineProperty, Ur = Object.getOwnPropertyDescriptor, Ct = (n, t, e, r) => {
  for (var a = r > 1 ? void 0 : r ? Ur(t, e) : t, i = n.length - 1, o; i >= 0; i--)
    (o = n[i]) && (a = (r ? o(t, e, a) : o(a)) || a);
  return r && a && Dr(t, e, a), a;
};
let rt = class extends x {
  constructor() {
    super(...arguments), this.format = "HH:mm:ss", this.tickInterval = 1e3, this.formatTime = (n) => {
      try {
        return xr(n, this.format);
      } catch {
        return /* @__PURE__ */ new Date();
      }
    }, this.dateNow = this.formatTime(/* @__PURE__ */ new Date()), this.setTicker = () => {
      this.ticker = setInterval(this.updateTime, this.tickInterval);
    }, this.updateTime = () => {
      this.dateNow = this.formatTime(/* @__PURE__ */ new Date()), this.requestUpdate();
    };
  }
  updated(n) {
    super.updated(n), n.size > 0 && n.get("tickInterval") !== void 0 && n.get("tickInterval") !== this.tickInterval && (clearInterval(this.ticker), this.setTicker());
  }
  connectedCallback() {
    super.connectedCallback(), this.setTicker();
  }
  disconnectedCallback() {
    super.disconnectedCallback(), console.log("disconnectedCallback"), clearInterval(this.ticker);
  }
  render() {
    return zt`<span>${this.dateNow}</span>`;
  }
};
Ct([
  Q({ type: String })
], rt.prototype, "format", 2);
Ct([
  Q({ type: Number })
], rt.prototype, "tickInterval", 2);
rt = Ct([
  Bt("counting-clock")
], rt);
var Wr = Object.defineProperty, Nr = Object.getOwnPropertyDescriptor, it = (n, t, e, r) => {
  for (var a = r > 1 ? void 0 : r ? Nr(t, e) : t, i = n.length - 1, o; i >= 0; i--)
    (o = n[i]) && (a = (r ? o(t, e, a) : o(a)) || a);
  return r && a && Wr(t, e, a), a;
};
let N = class extends x {
  constructor() {
    super(...arguments), this.href = null, this.onClick = () => {
    }, this.size = "default";
  }
  render() {
    return zt`
            <button class="custom-button ${this._getSize()}" @click=${this._onClick}>
                <slot>
                    Det hr borde inte synnas
                </slot>
            </button>
        `;
  }
  _onClick() {
    var n;
    this.href && ((n = this.href) == null ? void 0 : n.length) > 0 && (window.location.href = this.href), this.onClick && this.onClick();
  }
  _getSize() {
    switch (this.size) {
      case "small":
        return "button-small";
      case "large":
        return "button-large";
      default:
        return "default";
    }
  }
};
N.styles = re`
        .custom-button {
            font-family: Helvetica;
            font-size: 19px;
            font-weight: bold;
            box-shadow: 0 2px 10px 0 rgba(0,0,0,.33);
            border-radius: 7px;
            text-transform: uppercase;
            padding: 21px 54px;
            color: #fff;
            background-image: 
                linear-gradient(
                    -90deg,
                    #dd0890 0,
                    #200c9c 30%,
                    #4a0082 70%,
                    #dd0890 100%
                );
            cursor: pointer;
            background-size: 300% 100%;
            transition: all .4s ease-in-out;
            border: 0;
        }

        .custom-button.button-small {
            font-size: 16px;
            padding: 17px 45px;
        }

        .custom-button.button-large {
            font-size: 22px;
            padding: 25px 60px;
        }

        .custom-button:hover{
            background-position: 100% 0;
            -webkit-transition: all .4s ease-in-out;
            -o-transition: all .4s ease-in-out;
            transition: all .4s ease-in-out;
        }
    `;
it([
  Q({ type: String })
], N.prototype, "href", 2);
it([
  Q({ type: Function })
], N.prototype, "onClick", 2);
it([
  Q({ type: String })
], N.prototype, "size", 2);
N = it([
  Bt("nti-button")
], N);
export {
  rt as CountingClock,
  N as NTIButton
};
